### DISCUSSION TITLE
Inso support for Kong 3.x

### DISCUSSION THREAD

**Discussion by shawnchapla-8451** [Kong/insomnia#5383] [Help] [answered]
We've just begun testing Inso as a possible automated means to get from OpenAPI documents to Kong configuration in our 3.0 environment.  Inso successfully generates the declarative config file from the document, but when trying to use decK to apply it to Kong, we get this error:

```
Error: cannot apply '1.1' config format version to Kong version 3.0 or above.
Please upgrade your configuration to account for 3.0
breaking changes using the following command:

deck convert --from kong-gateway-2.x --to kong-gateway-3.x

This command performs the following changes:
  - upgrade the `_format_version` value to `3.0`
  - add the `~` prefix to all routes' paths containing a regex-pattern

These changes may not be correct or exhaustive enough.
It is strongly recommended to perform a manual audit
of the updated configuration file before applying
the configuration in production. Incorrect changes will result in
unintended traffic routing by Kong Gateway.

For more information about this and related changes,
please visit: https://docs.konghq.com/deck/latest/3.0-upgrade
```

Using ```deck convert``` does seem to fix it, but is there something we should be doing with inso to have it generate 3.x compatible decK, or doesn't it support that version of Kong yet?  Inso and decK versions are 3.6.0 and 1.15.1 respectively.

**Comment by wongstein** [comment]
Hi, in our documentation on Inso-CLI we document that Kong Gateway 3.X is not supported at this moment.  This is work that on our roadmap so stay tuned for when that support is completed.

https://docs.insomnia.rest/inso-cli/introduction

I'm going to keep this issue open so that the team can updates as they come.

**Reply by shawnchapla-8451** [reply]
OK, thanks.  I had to imagine that 3.x support wouldn't be ignored, but good to get confirmation that it is in the works.

**Comment by filfreire** [comment, marked as answer]
Kong 3.x support is being added - see PR https://github.com/Kong/insomnia/pull/5827
